runtime/fiber.c

```diff
@@ -613,12 +619,18 @@ void caml_maybe_expand_stack (void)
     (value*)stk->sp - Stack_base(stk);
   uintnat stack_needed =
     Stack_threshold / sizeof(value)
-    + 10 /* for words pushed by caml_start_program */;
-  /* XXX does this "8" need updating?  Provisionally changed to 10 */
+    /* for words pushed by caml_start_program */
+    + 8 + Stack_padding_word;
```

Comment:
In light of the deleted comment: is this `8` definitely right now? Can we make it a defined constant and check it somehow? Is there a matching comment in caml_start_program to keep things in sync? Are there any other places which care about this number?

Reply:
`caml_start_program` only pushes 4 words (plus the padding word) on the OCaml stack, so 8 is more than sufficient unless there's some other constraint. @mshinwell do you know why it was changed to 10 at one point?

Technically `alloc_for_stack` also cares about this number as `wosize` must not be smaller than it, but that can't happen since the min stack size is either a page or 64 words. Agreed it'd be a lot better to keep this in sync automatically...